timelock-server-benchmark-client/src/main/java/com/palantir/atlasdb/timelock/benchmarks/benchmarks/AbstractBenchmark.java

```diff
@@ -0,0 +1,180 @@
+        } finally {
+            stopExecutors();
+        }
+    }
+
+    private void scheduleTests() {
+        for (int i = 0; i < numClients; i++) {
+            executor.submit(this::runTestForSingleClient);
+        }
+    }
+
+    private void runTestForSingleClient() {
+        try {
+            recordTimesForSingleClient();
+        } catch (Throwable t) {
+            error = t;
+        }
+    }
+
+    private void recordTimesForSingleClient() {
+        for (int i = 0; i < requestsPerClient && error == null; i++) {
+            long duration = timeOneCall();
+            times.set(counter.getAndIncrement(), duration);
+            completionLatch.countDown();
+        }
+    }
+
+    private void waitForTestsToComplete() throws InterruptedException {
+        while (!completionLatch.await(1, TimeUnit.SECONDS)) {
+            throwIfAnyCallsFailed();
+            log.info("completed {} calls", counter.get());
+        }
+    }
+
+    private void throwIfAnyCallsFailed() {
+        if (error != null) {
+            throw Throwables.propagate(error);
+        }
+    }
+
+    private void stopExecutors() {
+        executor.shutdownNow();
+    }
+
+    private long timeOneCall() {
+        long start = System.nanoTime();
+        performOneCall();
+        long end = System.nanoTime();
+
+        return end - start;
+    }
+
+    protected abstract void performOneCall();
+
+    protected void setup() { }
+
+    protected Map<String, Object> getExtraParameters() {
+        return ImmutableMap.of();
+    }
+
+    private Map<String, Object> getStatistics() {
+        sortTimes();
+
+        Map<String, Object> result = Maps.newHashMap();
+        result.put("numClients", numClients);
+        result.put("requestsPerClient", requestsPerClient);
+        result.put("average", getAverageTimeInMillis());
+        result.put("p50", getPercentile(0.5));
+        result.put("p95", getPercentile(0.95));
+        result.put("p99", getPercentile(0.99));
+        result.put("totalTime", totalTime / 1_000_000.0);
+        result.put("throughput", getThroughput());
+        result.put("name", getClass().getSimpleName());
+        result.putAll(getExtraParameters());
+
+        return result;
+    }
+
+    private void sortTimes() {
+        long[] result = new long[times.length()];
+        for (int i = 0; i < times.length(); i++) {
+            result[i] = times.get(i);
+        }
+        Arrays.sort(result);
+
+        sortedTimes = result;
+    }
+
+    private double getAverageTimeInMillis() {
+        return Arrays.stream(sortedTimes).average().getAsDouble() / 1_000_000f;
+    }
+
+    private double getPercentile(double percentile) {
+        double count = sortedTimes.length;
+        int index = (int)(count * percentile);
+        return sortedTimes[index] / 1_000_000.0;
+    }
+
+    public double getThroughput() {
+        return (double)(numClients * requestsPerClient) / (totalTime / 1_000_000_000.0);
```

Comment:
probably want to cast `numClients` to double first to avoid integer overflow.

`return (((double)numClients) * requestsPerClient) / (totalTime / 1_000_000_000.0);`